rts/Lua/LuaAtlasTextures.cpp

```diff
@@ -0,0 +1,104 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#include <sstream>
+
+#include "LuaTextures.h"
+#include "Rendering/GlobalRendering.h"
+#include "Rendering/Textures/TextureAtlas.h"
+#include "System/Log/ILog.h"
+#include "LuaAtlasTextures.h"
+
+void LuaAtlasTextures::Clear()
+{
+	for (auto& kv : textureAtlasMap)
+		spring::SafeDelete(kv.second);
+
+	textureAtlasMap.clear();
+}
+
+std::string LuaAtlasTextures::Create(const std::string& name, const int xsize, const int ysize, const int allocatorType)
+{
+	CTextureAtlas* atlas = new CTextureAtlas(allocatorType, xsize, ysize);
```

Comment:
TODO

Reply:
> which you can do because every atlas stores its own name

Atlas `name != idStr` so update of textureAtlasMap upon .back() to deleted place position movement cannot be done nicely.
Are you sure you can't stand grandma-style C++? Because getting rid of it here proved tedious.

Reply:
Shouldn't be that tedious, `textureAtlasMap` just needs to map from `id`'s to indices (rather than from names) and `CTextureAtlas` extended with a `std::string id` member. The second part can even be skipped if you change the `CreateTextureAtlas` API to not accept user-chosen names, which are mostly irrelevant anyway.

Reply:
Have a look: https://github.com/spring/spring/pull/559/commits/57db9665e4916480031aa716c848ccbe30beee78